lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -218,11 +244,8 @@ def acquired(self) -> bool:  # noqa: C901
         Returns:
             Whether lock was acquired
         """
-        if self._opensearch.is_node_up():
-            host = self._charm.unit_ip
-        else:
-            host = None
-        alt_hosts = [host for host in self._charm.alt_hosts if self._opensearch.is_node_up(host)]
+        host = self._charm.unit_ip if self._opensearch.is_node_up() else None
+        alt_hosts = self._charm.alt_hosts
         if host or alt_hosts:
             logger.debug("[Node lock] 1+ opensearch nodes online")
             try:
```

Comment:
The logic below is the following:
```
                online_nodes = len(
                    ClusterTopology.nodes(
                        self._opensearch, use_localhost=host is not None, hosts=alt_hosts
                    )
                )
```

That means the main orchestrator may try to use the index lock  even if no data nodes are visible. We need a way to check how many nodes with the index lock are online, not the total amount of nodes.

On the other hand, non-data nodes could benefit of using lock only for service stop. So, a cluster_manager should stop itself only if: (1) all other peers in the same app are online; and (2) lock is free. At service start, we simply restart right away. That would reduce lock contention: we have one single index for the whole deployment.

Reply:
That's a good point - this should be addressed as part of the sec. index delayed initialization work.

Reply:
Okay, I will leave it open but this comment is resolved for this PR.